rust/protocol/src/storage/traits.rs

```diff
@@ -41,35 +76,43 @@ pub trait IdentityKeyStore {
         ctx: Context,
     ) -> Result<bool>;
 
+    /// Return the public identity for the given `address`, if known.
     async fn get_identity(
         &self,
         address: &ProtocolAddress,
         ctx: Context,
     ) -> Result<Option<IdentityKey>>;
 }
 
+/// Interface for storing pre-keys downloaded from a server.
 #[async_trait(?Send)]
 pub trait PreKeyStore {
+    /// Look up the pre-key from this store corresponding to `prekey_id`.
     async fn get_pre_key(&self, prekey_id: PreKeyId, ctx: Context) -> Result<PreKeyRecord>;
 
+    /// Set the entry for `prekey_id` to the value of `record` in this store.
```

Comment:
Nitpick: I think the "from this store" and "in this store" are redundant in most of these accessors.

Reply:
Absolutely right!